packages/runtime/src/useCases/transport/relationshipTemplates/CreateTokenForOwnRelationshipTemplate.ts

```diff
@@ -51,6 +87,10 @@ export class CreateTokenForOwnTemplateUseCase extends UseCase<CreateTokenForOwnT
             return Result.fail(RuntimeErrors.relationshipTemplates.personalizationMustBeInherited());
         }
 
+        if (template.passwordProtection && !template.passwordProtection.matchesInputForNewPasswordProtection(request.passwordProtection)) {
+            return Result.fail(RuntimeErrors.relationshipTemplates.passwordProtectionMustBeInherited());
+        }
```

Comment:
Hm, a more broader question that get's in my head while reading this:

Theoretically it's possible now to put a passwordProtected template inside of a non-protected token

1. do we want that?
2. will we run into issues with that? I don't really know how to handle this..

Reply:
`matchesInputForNesPasswordProtection` will return false if the new password protection is undefined - and for a protected template inside an unprotected token, the reference from the token would not reveal that a password is required. Hence I want to block that

Reply:
Oh, I misunderstood this. Looks good then.